core/src/main/scala/kafka/admin/ConfigCommand.scala

```diff
@@ -150,6 +159,115 @@ object ConfigCommand {
       Seq.empty
   }
 
+  case class Entity(entityType: String, sanitizedName: Option[String]) {
+    val entityPath = sanitizedName match {
+      case Some(n) => entityType + "/" + n
+      case None => entityType
+    }
+    override def toString: String = {
+      val typeName = entityType match {
+        case ConfigType.User => "user-principal"
+        case ConfigType.Client => "client-id"
+        case ConfigType.Topic => "topic"
+        case t => t
+      }
+      sanitizedName match {
+        case Some(ConfigEntityName.Default) => "default " + typeName
+        case Some(n) =>
+          val desanitized = if (entityType == ConfigType.User) QuotaId.desanitize(n) else n
+          s"$typeName '$desanitized'"
+        case None => entityType
+      }
+    }
+  }
+
+  case class ConfigEntity(root: Entity, child: Option[Entity]) {
+    val fullSanitizedName = root.sanitizedName.getOrElse("") + child.map(s => "/" + s.entityPath).getOrElse("")
+
+    def getAllEntities(zkUtils: ZkUtils) : Seq[ConfigEntity] = {
+      // Describe option examples:
+      //   Single entity name:
+      //     --entity-type topics --entity-name topic1 (topic1)
+      //     --entity-type users --entity-name user1 --entity-type clients --entity-name client2 (<user1, client2>)
```

Comment:
This doesn't seem to be a single entity as the comment in line 189 suggests? Ditto for line 193.